Recycler View implementation

aleksamarkoni opened the pull request:
Added the Recycler View, and appropriate adapter that extends it.
Updated the Loader to work with new adapter.

Added the appropriate onItemClick listener to ViewHolder. It now
informes the NewsFeedListActivity when there is a onClick of some
of the items in the list, after which NewsFeedListActivity display
detailed information about the news.

Small changes when saving the files.

aleksamarkoni committed:
Recycler View implementation

Added the Recycler View, and appropriate adapter that extends it.
Updated the Loader to work with new adapter.

Added the appropriate onItemClick listener to ViewHolder. It now
informes the NewsFeedListActivity when there is a onClick of some
of the items in the list, after which NewsFeedListActivity display
detailed information about the news.

Small changes when saving the files.

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedListFragment.java:
```diff
@@ -83,14 +93,17 @@ public void onActivityCreated(Bundle savedInstanceState) {
+
+    @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
```
Seems useless method. Contains only call for super method.

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedListFragment.java:
```diff
@@ -113,6 +126,7 @@ public void onDetach() {
         mCallbacks = sDummyCallbacks;
     }
 
+    /*
```
Please remove commented code if you are not going to uncomment it later

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedListFragment.java:
```diff
@@ -121,13 +135,14 @@ public void onListItemClick(ListView listView, View view, int position, long id)
+    */
 
     @Override
     public void onSaveInstanceState(Bundle outState) {
```
Does nothing special

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedListFragment.java:
```diff
@@ -138,16 +153,16 @@ public void onSaveInstanceState(Bundle outState) {
     public void setActivateOnItemClick(boolean activateOnItemClick) {
```
Does nothing

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedListFragment.java:
```diff
@@ -138,16 +153,16 @@ public void onSaveInstanceState(Bundle outState) {
     }
 
     private void setActivatedPosition(int position) {
         if (position == ListView.INVALID_POSITION) {
```
Empty if-else block

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsListAdapter.java:
```diff
@@ -1,50 +1,102 @@
-        } else {
-            view = convertView;
+    public void setData(List<SingleNewsItem> newData) {
+        newsItemList.clear();
```
You will clear list though newData can be null. In this case you will show nothing for user

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/components/FileStorage.java:
```diff
@@ -9,16 +11,50 @@
 
+    private static final String TAG = FileStorage.class.getSimpleName();
+
     Context context;
```
How about to make this field private?

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/components/FileStorage.java:
```diff
@@ -9,16 +11,50 @@
+            if (file == null) {
+                Log.e(TAG, "We cant open external files dir");
+            } else {
+                f = new File(context.getExternalFilesDir(null), filename);
```
file == context.getExternalFilesDir(null)

aleksamarkoni committed:
Adding the Search View and Material Toolbar

Both activities have the Toolbar as the ActionBar. First activity,
includes the menu in actionbar, and the second has HomeAsUp button
enabled.

SearchView added to the list activity, and logic for the
searching is now working. There is still things left to do.

Loader changed to support searching of the data based on text
that user enteres in the SearchView. We only search for the words
in the Title of the news.

akorshak commented on markovic/YahooNewsFeed/app/build.gradle:
```diff
@@ -23,4 +23,6 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'com.android.support:appcompat-v7:22.2.0'
     compile 'com.android.support:support-v4:22.2.0'
+    compile 'com.android.support:recyclerview-v7:21.0.0'
```
I think you can use  "compile 'com.android.support:design:22.2.0'" instead of 'com.android.support:recyclerview-v7:21.0.0' and 'com.android.support:appcompat-v7:22.2.0'. Please try.

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedDetailActivity.java:
```diff
@@ -16,15 +19,38 @@
+    /**
+     * Toolbar that is a replacement for the ActionBar
+     */
+    Toolbar toolbar;
```
Seems can be private

akorshak commented on markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsFeedDetailActivity.java:
```diff
@@ -16,15 +19,38 @@
-        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        toolbar = (Toolbar) findViewById(R.id.tool_bar);
+
+        if (toolbar != null) {
```
Are you sure that toolbar can be null here?

aleksamarkoni committed:
Card Layout, Picasso library, and WebView in Detail

1) Added Picasso image loading library to the project.

2) For every item in the list of news, we will use Material Design
Card Layout.

3) Implemented WebView in Detail Activity, so we can display Single
Image in this Activity. Also implemented progress bar to display info
about news loading progress.

4) Implemented Pull to Refresh option so now, when the user wants to
see if there is new news on the site, he can easily do so, with
simple swipe on the top of the list.
